### ISSUE TITLE
windows resized after rotation

### ISSUE THREAD

**Issue by kurka** [paperwm/PaperWM#305]
I am using paperWM on a 2-in-1 pc, which supports window rotation events.

Everything works fine when the screen is rotated, but I noticed that the dimension of the windows are lost everytime.

For instance, if I am using the pc on landscape, then rotate it to portrait, and back to landscape, all the windows are resized to a smaller width than they were originally in the landscape mode. 

Is there a way that the layout remains fixed either in portrait or landscape modes?

**Comment by hedning** [comment]
Yeah, there's room for improvement when handling changing monitor configurations. Eg. preserve our pseudo maximize.

> Is there a way that the layout remains fixed either in portrait or landscape modes?

You'll have to look up how to do this in gnome-shell/mutter. We only react to changes in the monitor configuration which I'm guessing gnome-shell/mutter triggers when they see the screen rotation change.

**Comment by Thesola10** [comment]
I think it's related to the fact that GNOME doesn't want windows to be wider than the screen width. In my experience, when rotating display, only windows wider than the new available width get resized.

You can try adding a "useful window width" to quickly restore the previous layout.
